app/controllers/api/v2/advisor_engine/advisor_engine_controller.rb

```diff
@@ -17,6 +22,72 @@ def host_details
             end
           end
         end
+
+        api :PATCH, "/advisor_engine/upload_hits", N_("Upload from insights advisor")
+        param :host_name, String, required: true
+        param :host_uuid, String
+
+        param :payload, Hash, :desc => N_("On prem payload including resolutions, rules, hits") do
+          param :resolutions, Array, :desc => N_("upload resolutions related to the hits") do
+            param :rule_id, String,  :desc => N_("rule id"), :required => true
+            param :description, String, :desc => N_("resolution description")
+            param :needs_reboot, :bool, :desc => N_("need reboot")
+            param :resolution_risk, String, :desc => N_("resolution risk")
+            param :resolution_type, String, :desc => N_("type")
+          end
+
+          param :rules, Array, :desc => N_("upload rules related to the hits") do
+            param :rule_id, String, :desc => N_("rule id"), :required => true
+            param :description, String, :desc => N_("rule description")
+            param :category_name, String, :desc => N_("category name")
+            param :impact_name, String, :desc => N_("impact name")
+            param :summary, String, :desc => N_("summary")
+            param :generic, String, :desc => N_("generic")
+            param :reason, String, :desc => N_("reason")
+            param :total_risk, :number, :desc => N_("total risk")
+            param :reboot_required, :bool, :desc => N_("reboot required")
```

Comment:
Getting an error when the value on the client is nil. Can we use have it as a string and then use Foremancast bool to switch it over?

Reply:
Resolved